cli/loader_lib.c

```diff
@@ -161,11 +335,46 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
     }
 
     // Introspect to find our own path
-    const char * lib_dir = jl_get_libdir();
+    const char *lib_dir = jl_get_libdir();
 
     // Pre-load libraries that libjulia-internal needs.
     int deps_len = strlen(dep_libs);
-    char * curr_dep = &dep_libs[0];
+    char *curr_dep = &dep_libs[0];
+
+    void *cxx_handle;
+    int done_probe = 0;
+#if defined(_OS_LINUX_)
+    int do_probe = 1;
+    char *probevar = getenv("JULIA_PROBE_LIBSTDCXX");
+    if (probevar) {
+        if (strcmp(probevar, "1") == 0 || strcmp(probevar, "yes"))
+            do_probe = 1;
+        else if (strcmp(probevar, "0") == 0 || strcmp(probevar, "no"))
+            do_probe = 0;
+    }
+    if (do_probe) {
+        char *cxxpath = libstdcxxprobe();
+        if (cxxpath) {
+            cxx_handle = dlopen(cxxpath, RTLD_LAZY);
+            char *dlr = dlerror();
+            if (dlr) {
+                size_t buflen = strlen(cxxpath) + 2048;
+                char *errbuf = malloc(buflen);
+                int err_len = snprintf(errbuf, buflen, "Error, cannot load \"%s\"\n"
+                                                       "dlinfo() - %s\n", cxxpath, dlr);
+                write_wrapper(STDOUT_FILENO, errbuf, (size_t)err_len);
+                free(errbuf);
```

Comment:
There are already wrapper functions supplied for this
```suggestion
                jl_loader_print_stderr("ERROR: Unable to dlopen(cxxpath) in parent!\n");
                jl_loader_print_stderr3("Message: ", dlerr, "\n");
```

n.b. make sure `Message: ` has a space after it